packages/web/src/components/common/notification-list/NotificationList.tsx

```diff
@@ -10,11 +10,29 @@ import { TransactionGroupsType } from "@components/common/notification-button/No
 
 interface NotificationListProps {
   txsGroupsInformation: TransactionGroupsType[];
+  modalRef: React.RefObject<HTMLDivElement | null>;
+  modalOutSideClick: () => void;
 }
 
 const NotificationList: React.FC<NotificationListProps> = ({
   txsGroupsInformation,
+  modalRef,
+  modalOutSideClick,
 }) => {
+  useEffect(() => {
```

Comment:
Is there a reason to handle events from a Ref object passed from a NotificationList?

Being passed a Ref object creates a dependency on the parent component.

Reply:
The reason why the Ref object is passed to the NotificationList is that WalletConnectorButton, the parent component, must be included in the area of ​​Ref so that event bubbling can be prevented when a button is clicked while the NotificationList is open. ! Do you have any better ideas?

Reply:
How about handling the WalletConnectorButton event of the parent component?

Reply:
Removed dependency on parent component and gave event capturing option as 3rd argument to modal event listener. And with stopPropagation() I blocked the event that would come after it. How about this method?

Reply:
It's nice that it can be done on a component-by-component basis.